### PR TITLE
Add DiffEqBot Post

### PR DESCRIPTION
**Pull request by kanav99**
Hi! Recently we set up a benchmarking bot for JuliaDiffEq by the name of @DiffEqBot and me and @ChrisRackauckas thought it was a cool new feature to be published on Julialang Blog. Therefore requesting reviews for the blog I have prepared. Thanks!

### CODE REVIEW

File: blog/_posts/2019-06-18-diffeqbot.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+failed_job:
+  script:
+    - curl "http://endpoint_to_tell_that_report_failed?repo=${repo}&pr=${pr}&commit=${commit}"
+  when: on_failure
+```
+
+![](https://i.imgur.com/ORRPWx7.jpg)
+
+
+4. As soon as we make this commit on the `BenchmarkingRepo` on the branch, the Gitlab CI detects that this pushed branch has a config file in it and it goes wild. It knows now that it needs to run this script! It looks for availible Giltab Runners. We don't use shared runner as they are not suitable for benchmarking jobs. We have our own dedicated Gitlab Runner for this purpose.
```
Comment (mschauer) [inline blog/_posts/2019-06-18-diffeqbot.md]:
```suggestion
4. As soon as we make this commit on the `BenchmarkingRepo` on the branch, the Gitlab CI detects that this pushed branch has a config file in it and it goes wild. It knows now that it needs to run this script! It looks for availible Gitlab Runners. We don't use shared runner as they are not suitable for benchmarking jobs. We have our own dedicated Gitlab Runner for this purpose.
```

Comment (mschauer) [inline blog/_posts/2019-06-18-diffeqbot.md]:
```suggestion
4. As soon as we make this commit on the `BenchmarkingRepo` on the branch, the Gitlab CI detects that this pushed branch has a config file in it and it goes wild. It knows now that it needs to run this script! It looks for available Giltab Runners. We don't use shared runner as they are not suitable for benchmarking jobs. We have our own dedicated Gitlab Runner for this purpose.
```

File: blog/_posts/2019-06-18-diffeqbot.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+For collecting diagrams, all you have to do is to make a dictionary called `DIAGRAMS` and save all the plot references in that dictionary. For eg - 
+
+```julia
+DIAGRAMS = Dict()
+sol = solve(prob, Tsit5())
+DIAGRAMS["Fig1"] = plot(sol)
+```
+
+Right now, `DIAGRAMS` cannot be a recursive Dictionary, i.e. an element of the dictionary can only be a reference to the plot.
```
Comment (mschauer) [inline blog/_posts/2019-06-18-diffeqbot.md]:
```suggestion
Right now, `DIAGRAMS` cannot be a recursive Dictionary, i.e. an element of the dictionary can only be a reference to the plot.
```

File: blog/_posts/2019-06-18-diffeqbot.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
+---
+layout: post
+title:  Hello @DiffEqBot
+author: <a href="http://github.com/kanav99">Kanav Gupta</a>
+---
+
+Hi! Today we all got a new member to the DiffEq family. Say Hi to our own [DiffEqBot](https://github.com/DiffEqBot) - A Bot which helps run benchmarks and compares with the current master of a given package. It also generates and stores the Reports generated in a [repository](https://github.com/DiffEqBot/Reports). What's special about this is that it is completely stateless (no databases involved at all, just juggling between repositories!) and it has no exposed public URLs. Even though highly inspired by Nanosoldier, this has a completely unique workflow.
```
Comment (mschauer) [inline blog/_posts/2019-06-18-diffeqbot.md]: >
```suggestion
Hi! Today we all got a new member to the DiffEq family. Say hi to our own [DiffEqBot](https://github.com/DiffEqBot) - A bot which helps run benchmarks and compares with the current master of a given package. It also generates and stores the Reports generated in a [repository](https://github.com/DiffEqBot/Reports). What's special about this is that it is completely stateless (no databases involved at all, just juggling between repositories!) and it has no exposed public URLs. Even though highly inspired by Nanosoldier, this has a completely unique workflow.
```

Comment (ChrisRackauckas) [reply blog/_posts/2019-06-18-diffeqbot.md]: Is there a change here?

Comment (kanav99) [reply blog/_posts/2019-06-18-diffeqbot.md]: Yeah I don't see any either

Comment (mschauer) [reply blog/_posts/2019-06-18-diffeqbot.md]: „Hi“ should be in quotation , just not German ones

Comment (mschauer) [reply blog/_posts/2019-06-18-diffeqbot.md]: One „Reports“ should be lower case

Comment (mschauer) [reply blog/_posts/2019-06-18-diffeqbot.md]: Bot should be lowercase

**Comment by ChrisRackauckas** [comment]
I will merge tomorrow morning.

**Comment by ViralBShah** [comment]
If the bot is more generally usable, perhaps it shouldn't be called DiffEqBot.

**Comment by fredrikekre** [comment]
> If the bot is more generally usable, perhaps it shouldn't be called DiffEqBot.

Related to that; it wasn't clear to me when reading the post if it was just for the DiffEq org or for general use.

**Comment by ChrisRackauckas** [comment]
It's just for the DiffEq org because we have the permissions setup only for a few of us on our repos. It's too invasive to just let anyone control it, and it uses our private gitlab runners.

But if someone wants to make use of it, just contact me and we can discuss it. I hope something like CuArrays and Flux get setup on it, and we'll have GPU benchmarking servers setup once we get a new GPU to slap into a computer that's currently doing nothing. @vchuravy might want to remind me of this.

**Comment by fredrikekre** [comment]
So this post just a user manual for DiffEq developers then? Perhaps not of interest for the audience of the julialang.org blog then.

**Comment by ViralBShah** [comment]
I don't think it is only for DiffEq developers only. It is just that a lot of infrastructure is set up only for DiffEq but others could also use it.

**Comment by ChrisRackauckas** [comment]
We already know it's of the interest of Julialang blog readers because details of this service came by more than one request  (we gathered names of people who wanted to know how to do this before for one of the NumFOCUS grants), so it's quite silly to debate it.

But do you really think that nobody else wants to setup performance regression testing for a Julia package? This post gives you step by step details on how to set one up.

**Comment by fredrikekre** [comment]
I was just asking since
> it wasn't clear to me when reading the post if it was just for the DiffEq org or for general use.

**Pull request merged**